frontend/src/hooks/queryBuilder/useTag.ts

```diff
@@ -61,13 +64,14 @@ export const useTag = (
 	}, []);
 
 	useEffect(() => {
-		setTags(
-			(query?.tagFilters?.items || []).map((obj) =>
-				isInNotInOperator(obj.op)
-					? `${obj.key} ${obj.op} ${obj.value.join(',')}`
-					: `${obj.key} ${obj.op} ${obj.value.join(' ')}`,
-			),
+		const initialTags = (query?.filters?.items || []).map((obj) =>
+			isInNInOperator(getOperatorFromValue(obj.op))
+				? `${obj.key?.key} ${getOperatorFromValue(
+						obj.op,
+				  )} ${(obj.value as string[]).join(', ')}`
```

Comment:
Among other issues, the commas are only a small part, and here is such an issue. The naive join breaks. say the value is `["a", "b,c", "d"]`, the `.join` will create `'a, b,c, d'` which in turn will create a value as `['a', ' b', 'c', ' d']` which is incorrect. You should use `unparse` with `{data: value}`. When you write a code, take a moment to think about what your doing and how it breaks.

Reply:
@srikanthccv - Removed join from value
https://github.com/SigNoz/signoz/pull/2629/commits/7517cbf499a5d2e9f22add2b1ac77d30f2bd90bf